pymeasure/display/console.py

```diff
@@ -0,0 +1,272 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+import copy
+import argparse
+
+try:
+    import progressbar
+    # Check that progressbar is progressbar2
+    progressbar.streams
+except (AttributeError, ImportError):
+    progressbar = None
+from .Qt import QtCore
+import signal
+from ..log import console_log
+
+from .browser import BaseBrowserItem
+from .manager import BaseManager, Experiment
+
+from ..experiment import Results, Procedure, unique_filename
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class ConsoleBrowserItem(BaseBrowserItem):
+
+    def __init__(self, progress_bar):
+        self.bar = progress_bar
+
+    def setStatus(self, status):
+        if self.bar:
+            self.bar.update(status=self.status_label[status])
+
+    def setProgress(self, progress):
+        if self.bar:
+            self.bar.update(progress)
+
+
+class ConsoleArgumentParser(argparse.ArgumentParser):
+    special_options = {
+        "no-progressbar":   {"default": False,
+                             "desc": "Disable progressbar",
+                             "help_fields": ["default"],
+                             "action": 'store_true'},
+        "log-level":        {"default": 'INFO',
+                             "choices": list(logging._nameToLevel.keys()),
+                             "desc": "Set log level (logging module values)",
+                             "help_fields": ["default"]},
+        "sequence-file":    {"default": None,
+                             "desc": "Sequencer file",
```

Comment:
As you're referring to the help string of the console mode (which is nice, since it helps keeping the tutorial concise), I'd suggest slightly expanding this string.

```suggestion
                             "desc": "Sequencer file as used/defined by the sequencer widget to "
                                     "execute a sequence of measurements",
```